contrib/backporting/cherry-pick

```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
+set -e
+if [ $# -ne 1 ]; then
+  echo "Usage: $0 <commit-id>"
+  exit 1
+fi
+CID=$1
+REM=`git remote -v | grep "github.com/cilium/cilium.git" | head -n1 | cut -f1`
+if [[ "`git branch -q -r --contains $CID $REM/master 2> /dev/null`" != *"$REM/master"* ]]; then
+  echo "Commit $CID not in $REM/master!"
+  exit 1
+fi
+TMPF1=`mktemp cp.XXXXXX`
+TMPF2=`mktemp cp.XXXXXX`
+FROM=`git show --pretty=email $CID | head -n 2 | grep "From: "`
+FULL_ID=`git show $CID | head -n 1 | cut -f 2 -d ' '`
+git show --patch-with-stat --pretty=email $FULL_ID > $TMPF1
+HDR_DONE=false
+OLD_IFS="$IFS"
+IFS=
+while read line
+do
+  echo "$line" >> $TMPF2
+  if [ "$line" == "" ]; then
```

Comment:
Somehow it seems like this line is generating a bunch of `unexpected operator` messages:

```
contrib/backporting/cherry-pick: 27: [: From 83547886793e1e0b9a132353da07a1e86b214693 Mon Sep 17 00:00:00 2001: unexpected operator
contrib/backporting/cherry-pick: 27: [: From: Romain Lenglet <romain.lenglet@berabera.info>: unexpected operator         
contrib/backporting/cherry-pick: 27: [: Date: Thu, 7 Dec 2017 01:06:58 -0800: unexpected operator                                                    
contrib/backporting/cherry-pick: 27: [: Subject: [PATCH] vendor: Upgrade github.com/optiopay/kafka (#2280): unexpected operator                                
contrib/backporting/cherry-pick: 27: [: unexpected operator                                                                                                             
contrib/backporting/cherry-pick: 27: [: Signed-off-by: Romain Lenglet <romain@covalent.io>: unexpected operator
contrib/backporting/cherry-pick: 27: [: ---: unexpected operator                                            
contrib/backporting/cherry-pick: 27: [:  vendor.conf                                        |  2 +-: unexpected operator
contrib/backporting/cherry-pick: 27: [:  vendor/github.com/optiopay/kafka/proto/messages.go | 28 +++++++++++++---------: unexpected operator
contrib/backporting/cherry-pick: 27: [:  2 files changed, 18 insertions(+), 12 deletions(-): unexpected operator
contrib/backporting/cherry-pick: 27: [: unexpected operator                                                                                                        
contrib/backporting/cherry-pick: 27: [: diff --git a/vendor.conf b/vendor.conf: unexpected operator                 
contrib/backporting/cherry-pick: 27: [: index 2522666a78a6..d9189f37916e 100644: unexpected operator      
contrib/backporting/cherry-pick: 27: [: --- a/vendor.conf: unexpected operator                                                
contrib/backporting/cherry-pick: 27: [: +++ b/vendor.conf: unexpected operator 
...
```

Reply:
Ok, included your diff below. Thanks!